cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java

```diff
@@ -255,4 +258,19 @@ public DatasetAdmin call() throws Exception {
       }
     }
   }
+
+  private static UserGroupInformation getUgiForDataset(Impersonator impersonator, DatasetId datasetInstanceId)
+    throws IOException, NamespaceNotFoundException {
+    // for system dataset do not look up owner information in store as we know that it will be null.
+    // Also, this is required for CDAP to start, because initially we don't want to look up owner admin
+    // (causing its own lookup) as the SystemDatasetInitiator.getDataset is called when CDAP starts
+    UserGroupInformation ugi;
```

Comment:
We had not finished the discussion, is there a change in place which will stop the infinite loop at startup if we remove this?